spring-batch-samples/src/main/java/org/springframework/batch/samples/mongodb/DeletionJobConfiguration.java

```diff
@@ -38,37 +42,41 @@
 
 /**
  * This job will remove document "foo3" from collection "person_out" using
- * {@link MongoItemWriter#setDelete(boolean)}.
+ * {@link MongoItemWriter#setMode(MongoItemWriter.Mode)}.
  *
  * @author Mahmoud Ben Hassine
  */
+@Configuration
 @EnableBatchProcessing
+@Import(DataSourceConfiguration.class)
```

Comment:
I understand you added this to fix the error `Caused by: org.springframework.beans.factory.NoSuchBeanDefinitionException: No bean named 'dataSource' available`. But this is actually not up to date anymore since we added a MongoDb job repository in 5.2, which does not require a datasource at all. I will update the sample accordingly.